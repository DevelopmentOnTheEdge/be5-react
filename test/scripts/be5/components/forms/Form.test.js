import React          from 'react';
import renderer       from 'react-test-renderer';
import Form           from '../../../../../src/scripts/be5/components/forms/Form';
import ModalForm           from '../../../../../src/scripts/be5/components/forms/ModalForm';
import SubmitOnChangeForm           from '../../../../../src/scripts/be5/components/forms/SubmitOnChangeForm';


test('Form', () => {

  const  component = renderer.create(
    <Form value={json} frontendParams={{documentName: 'test'}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('ModalForm', () => {

  const  component = renderer.create(
    <ModalForm value={json} frontendParams={{documentName: 'test'}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('SubmitOnChangeForm', () => {

  const  component = renderer.create(
    <SubmitOnChangeForm value={json} frontendParams={{documentName: 'test'}} />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

const json = {

  "data": {
    "attributes": {
      "bean": {
        "values": {
          "companyID": ""
        },
        "meta": {
          "/companyID": {
            "displayName": "companyID"
          }
        },
        "order": [
          "/companyID"
        ]
      },
      "entity": "companies",
      "layout": {},
      "operation": "SelectCompany",
      "operationResult": {
        "status": "open"
      },
      "query": "Selection view SelectCompany",
      "selectedRows": "",
      "title": "SelectCompany"
    },
    "type": "form"
  },
  "links": {
    "self": "form/companies/Selection view SelectCompany/SelectCompany"
  },
  "meta": {
    "_ts_": "1507198229686"
  }

};