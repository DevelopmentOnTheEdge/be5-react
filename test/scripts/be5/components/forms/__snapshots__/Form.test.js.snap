// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form 1`] = `
<div
  className="row"
>
  <div
    className="formBox col-12 max-width-970 formBoxDefault"
  >
    <h1
      className="form-component__title"
    >
      SelectCompany
    </h1>
    <form
      className=""
      id="1507198229686"
      onSubmit={[Function]}
    >
      <div
        className="property-set form-row"
      />
      <div
        className="formActions"
      >
        <div>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            title=""
            type="submit"
          >
            Submit
          </button>
           
        </div>
      </div>
    </form>
    <br />
  </div>
  <div
    className="col-12"
  />
</div>
`;

exports[`Form errorOnApply 1`] = `
<div
  className="row"
>
  <div
    className="formBox col-12 max-width-970 formBoxDefault"
  >
    <h1
      className="form-component__title"
    >
      Тестовая операция
    </h1>
    <form
      className=""
      id="1520488568877"
      onSubmit={[Function]}
    >
      <div
        className="property-set form-row"
      />
      <div
        className="formActions"
      >
        <div>
          <button
            className="btn btn-primary"
            disabled={false}
            onClick={[Function]}
            title=""
            type="submit"
          >
            Submit
          </button>
           
        </div>
      </div>
    </form>
    <br />
  </div>
  <div
    className="col-12"
  >
    <div
      className="errorPane"
    >
      <div
        className="errorPane__error"
      >
        <h1
          className="errorPane__title"
        >
          500
           - 
          Internal error occured during operation testtable.TestOperation
        </h1>
        <br />
        <pre
          className="errorPane__code"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Be5Exception: Internal error occured during operation... ",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`InlineForm 1`] = `
<form
  className="form-inline form-inline-mini"
  id="1507198229686"
  onSubmit={[Function]}
>
  <label
    className="mr-sm-2"
  >
    <strong>
      SelectCompany
    </strong>
  </label>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    title=""
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`ModalForm 1`] = `
<div>
  <div
    className="modal-header"
  >
    <h5
      className="modal-title"
    >
      SelectCompany
    </h5>
    <button
      aria-label="Close"
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
  </div>
  <form
    className="form-modal"
    id="1507198229686"
    onSubmit={[Function]}
  >
    <div
      className="modal-body"
    >
      <div
        className="property-set form-row"
      />
    </div>
    <div
      className="col-12"
    />
    <div
      className="modal-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        title=""
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;

exports[`SubmitOnChangeForm 1`] = `
<form
  className="submit-onchange-form"
  id="1507198229686"
>
  <input
    className="property-input form-control"
    id="companyIDPropertyInput"
    maxLength={undefined}
    onChange={[Function]}
    onInput={[Function]}
    onInvalid={[Function]}
    pattern={undefined}
    placeholder={undefined}
    required={true}
    size={undefined}
    type="text"
    value=""
  />
  <div
    className="col-12"
  />
</form>
`;
